### ISSUE TITLE
Add filter and search in Flask-admin for models with multiple foreignkey to same model fails

### ISSUE THREAD

**Issue by xqliu** [flask-admin/flask-admin#909]
This is a duplicate of stackoverflow post on http://stackoverflow.com/questions/25839330/add-filter-and-search-in-flask-admin-for-models-with-multiple-foreignkey-to-same

with more information, as I am also facing the same question / issue here.

Is there a way to add filter and search support in flask-admin for model that has 2 foreignkeys to the same model? An example sqlalchemy model is below - two of the fields are foreign keys to the same User model.

```
class Message(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    to_user_id = db.Column(db.Integer, db.ForeignKey('user.id'), index=True, nullable=False)
    from_user_id = db.Column(db.Integer, db.ForeignKey('user.id'), index=True, nullable=False)    
    subject = db.Column(db.String(512), default="")
    content = db.Column(db.Text, default="")
    to_user = relationship("User", foreign_keys=[to_user_id])
    from_user = relationship("User", foreign_keys=[from_user_id])
```

If I add 'to_user.id' to column_filters as you do for normal field, I get the following error because flask-admin doesn't know which fk fields to join User table on.

**InvalidRequestError: Could not find a FROM clause to join from. Tried joining to user, but got: Can't determine join between 'message' and 'user'; tables have more than one foreign key constraint relationship between them. Please specify the 'onclause' of this join explicitly.**

Another test case shown below:

```
class EnumValues(db.Model):
    __tablename__ = 'enum_values'
    id = Column(Integer, primary_key=True)
    type_id = Column(Integer, ForeignKey('enum_values.id'))
    type = relationship('EnumValues', remote_side=id,
                        backref=backref('type_values', lazy='dynamic'))
    code = Column(String(16), unique=True)
    display = Column(String(16), nullable=False)

    @staticmethod
    def type_filter(type_code):
        return AppInfo.get_db().session.query(EnumValues).\
            join(EnumValues.type, aliased=True).\
            filter_by(code=type_code)

class Incoming(db.Model):
    __tablename__ = 'incoming'
    id = Column(Integer, primary_key=True)
    amount = Column(Numeric(precision=8, scale=2, decimal_return_scale=2), nullable=False)
    date = Column(DateTime, nullable=False)

    category_id = Column(Integer, ForeignKey('enum_values.id'), nullable=False)
    category = relationship('EnumValues', foreign_keys=[category_id])

    status_id = Column(Integer, ForeignKey('enum_values.id'), nullable=False)
    status = relationship('EnumValues', foreign_keys=[status_id])

    @staticmethod
    def status_filter():
        return EnumValues.type_filter('INCOMING_STATUS')

    @staticmethod
    def type_filter():
        return EnumValues.type_filter('INCOMING_TYPE')

class ExpenseAdmin(ModelView):
    column_editable_list = ['date', 'amount', ]
    form_args = dict(
        status=dict(query_factory=Expense.status_filter),
        category=dict(query_factory=Expense.type_filter),
    )
    column_filters = ['date','amount','category','status']

    admin = Admin(app, u'Admin Home', base_template='layout.html', template_mode='bootstrap3')
    admin.add_view(ExpenseAdmin(Expense, db_session, menu_icon_type=ICON_TYPE_GLYPH, menu_icon_value='glyphicon-usd'))
```

**Comment by mrjoes** [comment]
Update to 1.2.0 and use string to specify the path: 'rel1.something'

**Comment by xqliu** [comment]
This is a duplicate of case #846. Closing it.

**Comment by xqliu** [comment]
Seems it's still not working if the filter is based on the same column of the join table,

in view.py, line 612 to 613, seems new join column will override the existing one as below.

```
            if joins:
                self._filter_joins[column] = joins  #Override and always contains only the latest filter defined.
            elif self._need_join(column.table):
                self._filter_joins[column] = [column.table]

            return flt

**Comment by mrjoes** [comment]
Ah, I think I understand.

So, problem is with the different filter paths that start from same column?

**Comment by xqliu** [comment]
Yes I think that' might has something to do with the issue, two filters were defined, one is based on category.display, another on status.display, both the two filter are join using enum_values.display, but using different foreign_key, (category_id vs status_id) in Incoming model definition.

And the field ___filter_joins__ only contains one element with key of __enum_values.display__, which one is left depends on exactly which one comes later in __column_filters__ list.

**Comment by petrus-jvrensburg** [comment]
Should be fixed in #1281
